app/database/operator/server_data_operator/handlers/post.test.ts

```diff
@@ -306,3 +310,100 @@ describe('*** Operator: Post Handlers tests ***', () => {
+    const databaseName = 'baseHandler.test.com';
+    const channelId = '1234';
+    beforeEach(async () => {
+        await DatabaseManager.init([databaseName]);
+        const serverDatabase = DatabaseManager.serverDatabases[databaseName]!;
+        database = serverDatabase.database;
+        operator = serverDatabase.operator;
+    });
+
+    afterEach(async () => {
+        await DatabaseManager.destroyServerDatabase(databaseName);
+    });
+
+    it('merge on empty chunks', async () => {
+        const newChunk = await transformPostsInChannelRecord({action: OperationType.CREATE, database, value: {record: undefined, raw: {channel_id: channelId, earliest: 0, latest: 100}}});
+        const chunks: PostsInChannelModel[] = [];
+        const result = await mergePostInChannelChunks(newChunk, chunks);
+        expect(result.length).toBe(0);
+        expect(newChunk.earliest).toBe(0);
+        expect(newChunk.latest).toBe(100);
+    });
+
+    it('remove contained chunks', async () => {
+        const newChunk = await transformPostsInChannelRecord({action: OperationType.CREATE, database, value: {record: undefined, raw: {channel_id: channelId, earliest: 0, latest: 100}}});
+        const chunks: PostsInChannelModel[] = [
+            await transformPostsInChannelRecord({action: OperationType.CREATE, database, value: {record: undefined, raw: {channel_id: channelId, earliest: 20, latest: 80}}}),
+        ];
+        const result = await mergePostInChannelChunks(newChunk, chunks);
+        expect(result.length).toBe(1);
+        expect(newChunk.earliest).toBe(0);
+        expect(newChunk.latest).toBe(100);
+        expect(result[0]).toBe(chunks[0]);
+        expect(chunks[0]._preparedState).toBe('destroyPermanently');
+    });
+
+    it('merge intersecting chunks', async () => {
+        const newChunk = await transformPostsInChannelRecord({action: OperationType.CREATE, database, value: {record: undefined, raw: {channel_id: channelId, earliest: 50, latest: 100}}});
+        const chunks: PostsInChannelModel[] = [
+            await transformPostsInChannelRecord({action: OperationType.CREATE, database, value: {record: undefined, raw: {channel_id: channelId, earliest: 25, latest: 70}}}),
+            await transformPostsInChannelRecord({action: OperationType.CREATE, database, value: {record: undefined, raw: {channel_id: channelId, earliest: 80, latest: 125}}}),
+        ];
+        const result = await mergePostInChannelChunks(newChunk, chunks);
+        expect(result.length).toBe(3);
+        expect(newChunk.earliest).toBe(25);
+        expect(newChunk.latest).toBe(125);
+        expect(newChunk._preparedState).toBe('update');
+        expect(result[0]).toBe(chunks[0]);
+        expect(chunks[0]._preparedState).toBe('destroyPermanently');
+        expect(result[1]).toBe(chunks[1]);
+        expect(chunks[1]._preparedState).toBe('destroyPermanently');
+        expect(result[2]).toBe(newChunk);
+        await operator.batchRecords(result, 'test');
```

Comment:
This batchRecords is used to silence an error on the database of preparing a model and not batching it.